core/src/main/java/io/grpc/transport/WritableBufferAllocator.java

```diff
@@ -32,13 +32,14 @@
 package io.grpc.transport;
 
 /**
- * The preferred way of creating a {@link WritableBuffer}.
+ * An allocator of buffers provided by the transport implementation to {@link MessageFramer} so
+ * it can send chunks of data to the transport in a form that the transport can directly serialize.
  */
 public interface WritableBufferAllocator {
 
   /**
-   * Allocate a new {@link WritableBuffer} that can hold
-   * {@code capacity} bytes.
+   * Request a new {@link WritableBuffer} with the given {@code capacity}. The allocator is
+   * free to return a buffer with a greater or lesser capacity.
```

Comment:
OK results of benchmark run are here

https://gist.github.com/louiscryan/c5997da99c51b5893a00

note that 'default' indicates the use of thread-local buffer caching. Some observations

- Pooled >>> Unpooled, particularly for larger buffers.
- Thread-local caching is masking the cost of arena locks which grow as buffer sizes grow, once we cross the 32k buffer cache boundary we see a moderate step in performance.
- After the 32k boundary allocation costs are relatively constant irrespective of buffer size until we exceed the area limit of 16MB. At which point performance absolutely stinks.

It seems like a toss-up between 4k & 16k  for the minimum size. For the maximum size we can either stop at 32k to limit to cacheable-buffers or we can go up to the arena size max. Given that the step-function around 32k is only 2x it seems reasonable to go higher, say 1MB to being with.